app/Domain/Clients/Controllers/ShowClient.php

```diff
@@ -118,37 +177,37 @@ public function run()
 
                 //Add comment
                 if (isset($_POST['comment']) === true) {
-                    if ($this->commentService->addComment($_POST, 'client', $id, $row)) {
+                    if ($this->commentService->addComment($_POST, 'client', $id)) {
                         $this->tpl->setNotification($this->language->__('notifications.comment_create_success'), 'success');
                     } else {
                         $this->tpl->setNotification($this->language->__('notifications.comment_create_error'), 'error');
                     }
                 }
 
-                if (isset($_POST['save']) === true) {
-                    $clientValues = [
-                        'id' => $row['id'],
-                        'name' => $_POST['name'],
-                        'street' => $_POST['street'],
-                        'zip' => $_POST['zip'],
-                        'city' => $_POST['city'],
-                        'state' => $_POST['state'],
-                        'country' => $_POST['country'],
-                        'phone' => $_POST['phone'],
-                        'internet' => $_POST['internet'],
-                        'email' => $_POST['email'],
+                if (isset($params['save']) === true) {
+                    $clientValues = (object) [
```

Comment:
Please map using the object. In fact since you already received the client object from the db above you can just use that and override the values with your post values., better yet would be to just do that in the service so that we can do validation in there.